lib/src/tiled_map.dart

```diff
@@ -0,0 +1,249 @@
+///   “even” or “odd” indexes along the staggered axis are shifted. (since 0.11)
+/// * backgroundcolor: The background color of the map. (optional, may include
+///   alpha value since 0.15 in the form #AARRGGBB. Defaults to fully transparent.)
+/// * nextlayerid: Stores the next available ID for new layers. This number is
+///   stored to prevent reuse of the same ID after layers have been removed.
+///   (since 1.2) (defaults to the highest layer id in the file + 1)
+/// * nextobjectid: Stores the next available ID for new objects. This number is
+///   stored to prevent reuse of the same ID after objects have been removed.
+///   (since 0.11) (defaults to the highest object id in the file + 1)
+/// * infinite: Whether this map is infinite. An infinite map has no fixed size
+///   and can grow in all directions. Its layer data is stored in chunks.
+///   (0 for false, 1 for true, defaults to 0)
+///
+/// The tilewidth and tileheight properties determine the general grid size of
+/// the map. The individual tiles may have different sizes. Larger tiles will
+/// extend at the top and right (anchored to the bottom left).
+///
+/// A map contains three different kinds of layers. Tile layers were once the
+/// only type, and are simply called layer, object layers have the objectgroup
+/// tag and image layers use the imagelayer tag. The order in which these layers
+/// appear is the order in which the layers are rendered by Tiled.
+///
+/// The staggered orientation refers to an isometric map using staggered axes.
+///
+/// Can contain at most one: <properties>
+///
+/// Can contain any number: <tileset>, <layer>, <objectgroup>, <imagelayer>,
+/// <group> (since 1.0), <editorsettings> (since 1.3)
+class TiledMap {
+  TileMapType type;
+  String version;
+  String? tiledVersion;
+
+  int width;
+  int height;
+  bool infinite;
+
+  int tileWidth;
+  int tileHeight;
+
+  List<Tileset> tilesets;
+  List<Layer> layers;
+
+  String? backgroundColor;
+  int compressionLevel;
+
+  int? nextLayerId;
+  int? nextObjectId;
+  MapOrientation? orientation;
+  RenderOrder renderOrder;
+
+  List<EditorSetting> editorSettings;
+  List<Property> properties;
+
+  // only for hexagonal maps:
+  int? hexSideLength;
+  StaggerAxis? staggerAxis;
+  StaggerIndex? staggerIndex;
+
+  TiledMap({
+    this.type = TileMapType.map,
+    this.version = "1.0",
+    this.tiledVersion,
+    required this.width,
+    required this.height,
+    this.infinite = false,
+    required this.tileWidth,
+    required this.tileHeight,
+    this.tilesets = const [],
+    this.layers = const [],
+    this.backgroundColor,
+    this.compressionLevel = -1,
+    this.hexSideLength,
+    this.nextLayerId,
+    this.nextObjectId,
+    this.orientation,
+    this.renderOrder = RenderOrder.rightDown,
+    this.staggerAxis,
+    this.staggerIndex,
+    this.editorSettings = const [],
+    this.properties = const [],
+  });
+
+  // Convenience Methods
+  Tile getTileByGid(int tileGid) {
```

Comment:
☝👇

Reply:
this is not part of the parser, so I can just change it to `tileByGid`